pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -763,3 +767,157 @@ func (os *OpenStack) GetZone() (cloudprovider.Zone, error) {
 func (os *OpenStack) Routes() (cloudprovider.Routes, bool) {
 	return nil, false
 }
+
+// Attaches given cinder volume to the compute running kubelet
+func (os *OpenStack) AttachDisk(diskName string) (string, error) {
+	disk, err := os.getVolume(diskName)
+	if err != nil {
+		return "", err
+	}
+	cClient, err := openstack.NewComputeV2(os.provider, gophercloud.EndpointOpts{
+		Region: os.region,
+	})
+	if err != nil || cClient == nil {
+		glog.Errorf("Unable to initialize nova client for region: %s", os.region)
+		return "", err
+	}
+	compute_id, err := os.getComputeIDbyHostname(cClient)
+	if err != nil || len(compute_id) == 0 {
+		glog.Errorf("Unable to get minion's id by minion's hostname")
+		return "", err
+	}
+
+	if len(disk.Attachments) > 0 && disk.Attachments[0]["server_id"] != nil {
+		if compute_id == disk.Attachments[0]["server_id"] {
+			glog.V(4).Infof("Disk: %q is already attached to compute: %q", diskName, compute_id)
+			return disk.ID, nil
+		} else {
+			errMsg := fmt.Sprintf("Disk %q is attached to a different compute: %q, should be detached before proceeding", diskName, disk.Attachments[0]["server_id"])
+			glog.Errorf(errMsg)
+			return "", errors.New(errMsg)
+		}
+	}
+	// add read only flag here if possible spothanis
+	_, err = volumeattach.Create(cClient, compute_id, &volumeattach.CreateOpts{
+		VolumeID: disk.ID,
+	}).Extract()
+	if err != nil {
+		glog.Errorf("Failed to attach %s volume to %s compute", diskName, compute_id)
+		return "", err
+	}
+	glog.V(2).Infof("Successfully attached %s volume to %s compute", diskName, compute_id)
```

Comment:
We should be consistent across plugins where possible. GCE PD is V(1) and I think successful attach and detach should be V(1). It makes it much easier to debug when we hit issues in the wild. And it doesn't happens often enough to clutter the logs. But, I don't feel strongly either way.